iTechArt.Repository/Repositories/GroceryRepository.cs

```diff
@@ -20,9 +21,9 @@ public GroceryRepository(AppDbContext dbContext, IMapper mapper)
         /// <summary>
         /// Get all entities from database.
         /// </summary>
-        public async Task<IGrocery[]> GetAllAsync()
+        public async Task<IGrocery[]> GetAllAsync(int pageIndex)
         {
-            return await _dbContext.Groceries.Select(grocery => _mapper.Map<Grocery>(grocery)).ToArrayAsync();
+            return await _dbContext.Groceries.Paginate(pageIndex).Select(groceries => _mapper.Map<Grocery>(groceries)).ToArrayAsync();
```

Comment:
groceries means many of grocery.
But in select statement it is one element.

Reply:
It is not my unit so did not make serious changes in structure

Reply:
you rename grocery to groceries